api/jobset/v1alpha2/jobset_types.go

```diff
@@ -84,6 +84,8 @@ type JobSetSpec struct {
 	// ReplicatedJobs is the group of jobs that will form the set.
 	// +listType=map
 	// +listMapKey=name
+	// +kubebuilder:validation:MinItems:=1
+	// +required
```

Comment:
Required fields should not be omitempty

Is this a new API or has it shipped? Changing a field to required is technically a breaking change

Reply:
It has shipped and is used. I was thinking of increasing this validation because we could not find any reason why someone would not set the replicated jobs. We hit some corner cases in validation that did not expect this.

im waiting on @ahg-g and @danielvegamyhre to see if we should do this for v1 alpha.

we could consider this when we promote the APIs to V1.

Reply:
Changes to fields being required are not ratcheted, so this could block writes to the object if anyone is using this and they haven't already got the field set to some value

Reply:
we can do that for v1 then.

Reply:
This is still alpha API. So fully backward compatibility is not guaranteed, right?
Additionally, do we really see situation where null replicatedJob? Isn't it indicate there are not any kind of workloads for JobSet?

Reply:
> This is still alpha API. So fully backward compatibility is not guaranteed, right?

yes, but then we need to do v1alpha3

Reply:
I am not sure that we should migrate API to the newer version when we limit number of ReplicatedJob.
Users should not create JobSet with no ReplicatedJob.
For me it looks more like a bug, not a feature.